mtscomp.py

```
@@ -804,23 +804,26 @@ def __del__(self):
-    for chunk_idx, chunk_start, chunk_length in tqdm(
-            unc.iter_chunks(), total=unc.n_chunks, desc='Checking'):
-        chunk = unc.read_chunk(chunk_idx, chunk_start, chunk_length)
-        # Find the corresponding chunk from the original data array.
-        i0, i1 = unc.chunk_bounds[chunk_idx], unc.chunk_bounds[chunk_idx + 1]
-        expected = data[i0:i1]
-        # Check the dtype and shape match.
-        assert chunk.dtype == expected.dtype
-        assert chunk.shape == expected.shape
-        if np.issubdtype(chunk.dtype, np.integer):
-            # For integer dtypes, check that the array are exactly equal.
-            assert np.array_equal(chunk, expected)
-        else:
-            # For floating point dtypes, check that the array are almost equal
-            # (diff followed by cumsum does not yield exactly the same floating point numbers).
-            assert np.allclose(chunk, expected, atol=CHECK_ATOL)
+    try:
+        # Read all chunks.
+        for chunk_idx, chunk_start, chunk_length in tqdm(
+                unc.iter_chunks(), total=unc.n_chunks, desc='Checking'):
+            chunk = unc.read_chunk(chunk_idx, chunk_start, chunk_length)
+            # Find the corresponding chunk from the original data array.
+            i0, i1 = unc.chunk_bounds[chunk_idx], unc.chunk_bounds[chunk_idx + 1]
+            expected = data[i0:i1]
+            # Check the dtype and shape match.
+            assert chunk.dtype == expected.dtype
+            assert chunk.shape == expected.shape
+            if np.issubdtype(chunk.dtype, np.integer):
+                # For integer dtypes, check that the array are exactly equal.
+                assert np.array_equal(chunk, expected)
+            else:
+                # For floating point dtypes, check that the array are almost equal
+                # (diff followed by cumsum does not yield exactly the same floating point numbers).
+                assert np.allclose(chunk, expected, atol=CHECK_ATOL)
```

Will this try block hide any exception raised in it? If so perhaps we should add a `logger.error()` in the `except` block?

---

Because there's no except block it will continue to raise any error after the finally block

---

ok!